lightning/src/ln/channel.rs

```diff
@@ -2193,6 +2193,18 @@ impl<Signer: WriteableEcdsaChannelSigner> Channel<Signer> {
 		self.context.channel_transaction_parameters.funding_outpoint.unwrap()
 	}
 
+	/// Returns whether an inbound HTLC on this channel has already been failed.
+	/// Returns None if the HTLC is not found.
+	pub fn check_inbound_htlc_failed(&self, htlc_id: u64) -> Option<bool> {
```

Comment:
Although in general they're handled, in tests I think they should get caught by the debug assert in `Channel::fail_htlc/get_update_fulfill_htlc` asserting that if we are claiming/failing back an HTLC that's no longer present, it must be already claimed (i.e. present in `self.context.historical_inbound_htlc_fulfills`).

Seeing that I imagined we always prevented duplicate fails, though now that I'm looking there again I see the comment saying in case of block rescans we handle them. So, is it fine to remove that debug assert, or is there a way to test this that works around it somehow?

Reply:
I just ended up removing the debug asserts